cypress/integration/multi_team/system_message_spec.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+/* eslint max-nested-callbacks: ["error", 4] */
+
+function verifySystemMessage(post) {
+    cy.get(post).
+        invoke('attr', 'class').
+        should('contain', 'post--system').
+        and('not.contain', 'same--root').
+        and('not.contain', 'other--root').
+        and('not.contain', 'current--user').
+        and('not.contain', 'post--comment').
+        and('not.contain', 'post--root');
+
+    cy.get(post).
+        find('.status-wrapper .status svg').
+        should('not.be.visible');
+}
+
+describe('System Message', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    const displayTypes = ['COMPACT', 'STANDARD'];
+
+    displayTypes.forEach((type) => {
+        it(`MM-15240 - no status on a system message with ${type} display`, () => {
+        // # Set message display
+            cy.changeMessageDisplaySetting(type);
+
+            // # Update the header to a long string
+            cy.getCurrentChannelId().then((channelId) => {
+                cy.apiUpdateChannel({
+                    id: channelId,
+                    header: ' Updating header',
+                }).then(() => {
+                    // # Get last post
+                    cy.getLastPostId().then((postId) => {
+                        cy.get(`#post_${postId}`).as('SystemMessage');
+                    });
```

Comment:
@saturninoabril @thekiiingbob These last two callbacks are 5 and 6 layers deep and ESLint doesn't like that. However, I'm not sure of a way around it. Maybe move the code in the `forEach` into a function?

Reply:
I'm okay as is with the `6` layers deep but if you're going to re-arrange the logic, one way is:
1) Patch the channel only once and move it in `before hook`
```javascript
before(() => {
    ...
    cy.getCurrentChannelId().then((channelId) => {
        const header = ' Updating header'.repeat(Math.floor(Math.random() * 10));
        cy.apiPatchChannel(channelId, {header});
    });
});
```
2) For each `it`, keep testing with `cy.getLastPostId` and `verifySystemMessage('@SystemMessage', type)`

3.  Note that I passed type into `verifySystemMessage` just to ensure that we're actually verifying for standard/compact message
```javascript
function verifySystemMessage(post, type) {
    if (type === 'compact') {
        cy.get(post).
            invoke('attr', 'class').
            should('contain', 'post--compact');
    }
    ...
});
```